packages/design-system-react/src/components/button-base/ButtonBase.test.tsx

```diff
@@ -177,4 +177,44 @@ describe('ButtonBase', () => {
     expect(button).toBeDisabled();
     expect(button).toHaveClass('opacity-50', 'cursor-not-allowed');
   });
+
+  it('handles text children correctly', () => {
+    // Test basic text wrapping and styling
+    const { rerender } = render(<ButtonBase>Click me</ButtonBase>);
+    const textElement = screen.getByText('Click me');
+
+    expect(textElement.tagName).toBe('SPAN');
+    expect(textElement).toHaveClass(
+      'text-inherit',
+      'text-s-body-md',
+      'font-s-body-md',
+      'leading-s-body-md',
+      'tracking-s-body-md',
+      'md:text-l-body-md',
+    );
+
+    // Test custom text props
+    rerender(
+      <ButtonBase textProps={{ className: 'custom-text-class' }}>
+        Click me
+      </ButtonBase>,
+    );
+    expect(screen.getByText('Click me')).toHaveClass('custom-text-class');
+
+    // Test non-string children
+    rerender(
+      <ButtonBase>
+        <div data-testid="custom-child">Custom Element</div>
+      </ButtonBase>,
+    );
+    const customChild = screen.getByTestId('custom-child');
+    expect(customChild.parentElement).not.toHaveClass(
+      'text-inherit',
+      'text-s-body-md',
+      'font-s-body-md',
+      'leading-s-body-md',
+      'tracking-s-body-md',
+      'md:text-l-body-md',
+    );
+  });
```

Comment:
Adding tests to verify the following:  

- `children` are correctly wrapped by the `Text` component.  
- The `textProps` prop functions as expected.  
- Custom elements are not rendered with the `Text` component when `children` is not a string.